elliottlib/cli/verify_attached_operators_cli.py

```diff
@@ -83,59 +131,51 @@ def _is_bundle(image_build):
-                image_specs[pullspec['new']] = image['nvr']
-        elif _is_appregistry(image):
-            for pullspec in _download_appregistry_image_references(image):
-                image_specs[pullspec] = image['nvr']
-    return image_specs
+            image['csv'] = _download_bundle_csv(image)
+            bundles.append(image)
+    return bundles
 
 
-def _download_appregistry_image_references(appregistry_build):
-    # for appregistry, image references are buried in the CSV in an archive
+def _download_bundle_csv(bundle_build):
+    # the CSV is buried in an archive
     url = constants.BREW_DOWNLOAD_TEMPLATE.format(
-        name=appregistry_build['package_name'],
-        version=appregistry_build['version'],
-        release=appregistry_build['release'],
+        name=bundle_build['package_name'],
+        version=bundle_build['version'],
+        release=bundle_build['release'],
         file_path="operator-manifests/operator_manifests.zip",
     )
     try:
         res = requests.get(url, timeout=10.0)
     except Exception as ex:
-        raise ElliottFatalError(f"appregistry data download {url} failed: {ex}")
+        raise ElliottFatalError(f"bundle data download {url} failed: {ex}")
     if res.status_code != 200:
-        raise ElliottFatalError(f"appregistry data download {url} failed (status_code={res.status_code}): {res.text}")
+        raise ElliottFatalError(f"bundle data download {url} failed (status_code={res.status_code}): {res.text}")
 
-    minor_version = re.match(r'^v(\d+\.\d+)', appregistry_build['version']).groups()[0]
     csv = {}
     with ZipFile(BytesIO(res.content)) as z:
         for filename in z.namelist():
-            if re.match(f"^{minor_version}/.*clusterserviceversion.yaml", filename):
+            if re.match(r"^.*clusterserviceversion.yaml", filename):
                 with z.open(filename) as csv_file:
                     if csv:
-                        raise ElliottFatalError(f"found more than one CSV in {appregistry_build['nvr']}?!? {filename}")
+                        raise ElliottFatalError(f"found more than one CSV in {bundle_build['nvr']}?!? {filename}")
                     csv = yaml.safe_load(csv_file)
 
     if not csv:
-        raise ElliottFatalError(f"could not find the csv for appregistry {appregistry_build['nvr']}")
-    return [ref['image'] for ref in csv['spec']['relatedImages']]
+        raise ElliottFatalError(f"could not find the csv for bundle {bundle_build['nvr']}")
+    return csv
 
 
 def _get_shipped_images(runtime, brew_session):
     # retrieve all image builds ever shipped for this version (potential operands)
     # NOTE: this will tend to be the slow part, aside from querying ET
     tag = f"{runtime.branch}-container-released"
-    tags = {tag, tag.replace('-rhel-7-', '-rhel-8-')}  # may be one or two depending
+    tags = {tag, tag.replace('-rhel-7-', '-rhel-8-')}  # set of one or two depending
```

Comment:
We will have rhel 9 images so this will probably be a problem.
But I am ok to live with this for now because that is the existing logic.
This could be improved by checking if a missing build is already shipped or not.

Reply:
yeah; i meant to update this but forgot about it, let me do that.

as we don't have rhel 7 anymore since 4.6, we can just update this to include rhel 9

ideally we would look these up from ocp-build-data, but i didn't want to pull that into this change...